pkg/actuators/machine/instances_test.go

```diff
@@ -1544,3 +1545,49 @@ func TestGetAvalabilityZoneTypeFromZoneName(t *testing.T) {
 		})
 	}
 }
+
+func TestGetCapacityReservationSpecification(t *testing.T) {
+	mockCapacityReservationID := "cr-1234a6789d234f6f4"
+	invalidMockCapacityReservationID := "cr-1234"
```

Comment:
Can you try a few variations, different lengths, ones with capitals, ones with non hex digits